File: MaxSliceSum.java
class MaxSliceSum {
    public int solution(int[] A) {
        int tMax = A.length > 0 ? A[0] : 0;
        int fMax = tMax;

        for (int i = 1; i < A.length; i++) {
            tMax = tMax < 0 ? 0 : tMax;
            tMax += A[i];

            if (tMax > fMax) {
                fMax = tMax;
            } 
        }

        return fMax;
    }

    public static void main(String[] args) {
        int[] a = new int[]{3, 2, -6, 4, 0};

        MaxSliceSum s = new MaxSliceSum();
        System.out.println(s.solution(a));
    }
}

File: MaxSliceSum.php
<?php

function solution($A) {
    $tMax = count($A) > 0 ? $A[0] : 0;
    $fMax = $tMax;

    $L = count($A);
    for ($i= 1; $i < $L; $i++) {
        $tMax = $tMax < 0 ? 0: $tMax;
        $tMax += $A[$i];

        if ($tMax > $fMax) {
            $fMax = $tMax;
        }
    }

    return $fMax;
}

$a = [3, 2, -6, 4, 0];
$a = [];
$a = [-10];
$a = [-2, 1];
$a = [-20, -30, -50];
$a = [-40, -30, -50];
var_dump(solution($a));